src/core/gui/MainWindow.cpp

```diff
@@ -101,8 +101,11 @@ MainWindow::MainWindow(GladeSearchpath* gladeSearchPath, Control* control, GtkAp
                      this);
 
     // "watch over" all key events
-    g_signal_connect(this->window, "key-press-event", G_CALLBACK(gtk_window_propagate_key_event), nullptr);
-    g_signal_connect(this->window, "key-release-event", G_CALLBACK(gtk_window_propagate_key_event), nullptr);
+    auto keyPropagate = +[](GtkWidget* w, GdkEvent* e, gpointer) {
```

Comment:
I don't understand why you use `GdkEvent* e` instead of `GdkEventKey* e` here, when the signature in the docs is 
```c
gboolean
key_release_event (
  GtkWidget* self,
  GdkEventKey event,
  gpointer user_data
)
```

Reply:
Ok, thanks for the explanations. Please go ahead and merge.